In [1]:
import matplotlib.pyplot as plt
import os
import pandas as pd
import sys
from time import time
import re
import datetime as dt

%matplotlib inline

%pylab inline
pylab.rcParams['figure.figsize'] = (20.0, 10.0)

%load_ext autoreload
%autoreload 2

# Constants definition
DATA_PATH = '../data/preprocessed_data/'
PROJECT_ROOT = '../'
SRC = PROJECT_ROOT + 'src/'

DATA = 'data.csv'

sys.path.append(SRC)

Populating the interactive namespace from numpy and matplotlib


In [13]:
data_df = pd.read_csv(DATA_PATH + DATA, index_col=0)
print(data_df.shape)
data_df.head()

(6120, 7)


exchange_rate symbol        date    cpi                         name  \
0        0.27224    AED  2007-01-15  86.95  United Arab Emirates dirham   
1        0.27230    AED  2007-02-15  86.95  United Arab Emirates dirham   
2        0.27242    AED  2007-03-15  86.95  United Arab Emirates dirham   
3        0.27233    AED  2007-04-15  86.95  United Arab Emirates dirham   
4        0.27241    AED  2007-05-15  86.95  United Arab Emirates dirham   

                country                      region  
0  United Arab Emirates  Middle East & North Africa  
1  United Arab Emirates  Middle East & North Africa  
2  United Arab Emirates  Middle East & North Africa  
3  United Arab Emirates  Middle East & North Africa  
4  United Arab Emirates  Middle East & North Africa

In [14]:
data_df['region'].unique()

array(['Middle East & North Africa', 'Europe & Central Asia',
       'Latin America & Caribbean', 'North America', 'East Asia & Pacific',
       'Sub-Saharan Africa', 'South Asia'], dtype=object)

In [16]:
data_df[data_df['region'] == 'Europe & Central Asia']['symbol'].unique()

array(['BGN', 'CHF', 'CYP', 'CZK', 'DKK', 'EUR', 'GBP', 'HRK', 'HUF',
       'ISK', 'NOK', 'PLN', 'RON', 'RUB', 'SEK', 'SKK', 'TRY'], dtype=object)

In [17]:
data_df[data_df['region'] == 'Europe & Central Asia']['name'].unique()

array(['Bulgarian Lev', 'Swiss Franc', 'Cyprus Pound', 'Czech Koruna',
       'Danish Krone', 'Euro', 'Pound Sterling', 'Croatian Kuna', 'Forint',
       'Iceland Krona', 'Norwegian Krone', 'Zloty', 'Romanian New Leu',
       'Russian Ruble', 'Swedish Krona', 'Slovak Koruna',
       'New Turkish Lira'], dtype=object)

## Many European currencies are "paired" to the Euro but they are allowed to fluctuate as much as 15%, and so, they have their own information. They will be kept.